### ISSUE TITLE
isTtySupported() does not take into account open_basedir() restrictions

### ISSUE THREAD

**Issue by LukeTowers** [symfony/symfony#54874]
### Symfony version(s) affected

`>= 6.2`

### Description

In https://github.com/symfony/symfony/commit/007fa61ffe703fcab9301226f5c31b6f87437137 / https://github.com/symfony/symfony/pull/46971 the logic that checks if TTY is supported changed from checking `/dev/tty` for reading / writing capabilities directly to checking `stream_isatty(\STDOUT)`. 

This causes an issue because the new way fails to take into account the open_basedir restrictions. 

Related: https://github.com/wintercms/storm/pull/175

### How to reproduce

`open_basedir` restriction not in effect, works fine:
```bash
php -r "require_once(__DIR__ . '/vendor/autoload.php'); (new \Symfony\Component\Process\Process(['ls', '-lsa']))->setTty(true)->run();"
```

`open_basedir` restriction in effect, causes hard crash with `PHP Fatal error:  Uncaught Symfony\Component\Process\Exception\RuntimeException: Unable to launch a new process. in vendor/symfony/process/Process.php:350`:
```bash
php -d open_basedir="/path/to/cwd" -r "require_once(__DIR__ . '/vendor/autoload.php'); (new \Symfony\Component\Process\Process(['ls', '-lsa']))->setTty(true)->run();"
```

### Possible Solution

Revert the linked commit or add an additional `is_writable('/dev/tty')` check.

### Additional Context

Side note, can we change the `self::isTtySupported()` call to `static::isTtySupported()` instead so that extending classes can override just the relevant method if needed in the future instead of having to override the `setTty()` method entirely?

**Comment by LukeTowers** [comment]
Ping @Yurunsoft

**Comment by mjauvin** [comment]
Fixed by #54863
